### DISCUSSION TITLE
Expose `previousState` and/or `frameData` in Frog.image Context

### DISCUSSION THREAD

**Discussion by sud0bug** [wevm/frog-archived#419] [General]
## Current Situation
Currently, the [Frog.image Context](https://frog.fm/reference/frog-image-context) only exposes `req`, `res`, and `var` objects. It does not receive other data such as `previousState` or `frameData` that the `Frog.frame` `Context` object receives.

## Problem
This limitation means  for image generation, developers need to:
1. Perform additional computations
2. Fetch data again
3. Pass data to the `image` route via URL parameters

These steps are often redundant, as the necessary operations have typically already been performed in the corresponding `frame` handler.

## Proposed Solution
Expose the same data in the `Frog.image` Context that is available in the `Frog.frame` Context. Specifically, include:
- `previousState`
- `frameData`
- Any other relevant data currently available only to `Frog.frame`

**Comment by dalechyn** [comment]
Thank you so much for this feature suggestion, this is awesome and I believe that we need to implement it!

**Comment by dalechyn** [comment]
It is now implemented and eased the frame development very much! Thanks for the suggestion!
